examples/apps/attributable-map/src/view.ts

```diff
@@ -0,0 +1,141 @@
+	wrapperDiv.appendChild(buttonsContainerDiv);
+
+	// Create red button element and add styling
+	const redButton = document.createElement("button");
+	redButton.style.fontSize = "2rem";
+	redButton.style.padding = "1rem 2rem";
+	redButton.style.borderRadius = "1rem";
+	redButton.style.backgroundColor = "#F84F31";
+	redButton.style.color = "white";
+	redButton.style.border = "none";
+	redButton.style.cursor = "pointer";
+	redButton.textContent = "Hit";
+	buttonsContainerDiv.appendChild(redButton);
+
+	redButton.addEventListener("click", () => {
+		hitCounter.hit("red");
+	});
+
+	// Create green button element and add styling
+	const greenButton = document.createElement("button");
+	greenButton.style.fontSize = "2rem";
+	greenButton.style.padding = "1rem 2rem";
+	greenButton.style.borderRadius = "1rem";
+	greenButton.style.backgroundColor = "#23C552";
+	greenButton.style.color = "white";
+	greenButton.style.border = "none";
+	greenButton.style.cursor = "pointer";
+	greenButton.textContent = "Hit";
+	buttonsContainerDiv.appendChild(greenButton);
+
+	greenButton.addEventListener("click", () => {
+		hitCounter.hit("green");
+	});
+
+	// Create container div for attribution text and add styling
+	const attributionContainerDiv = document.createElement("div");
+	attributionContainerDiv.style.display = "flex";
+	attributionContainerDiv.style.width = "100%";
+	attributionContainerDiv.style.justifyContent = "space-around";
+	attributionContainerDiv.style.marginBottom = "2rem";
+	wrapperDiv.appendChild(attributionContainerDiv);
+
+	const redAttributionContainerDiv = document.createElement("div");
+	redAttributionContainerDiv.style.fontSize = "2rem";
+	redAttributionContainerDiv.style.fontStyle = "italic";
+	redAttributionContainerDiv.style.textAlign = "center";
+	redAttributionContainerDiv.style.marginBottom = "2rem";
+	attributionContainerDiv.appendChild(redAttributionContainerDiv);
+
+	const greenAttributionContainerDiv = document.createElement("div");
+	greenAttributionContainerDiv.style.fontSize = "2rem";
+	greenAttributionContainerDiv.style.fontStyle = "italic";
+	greenAttributionContainerDiv.style.textAlign = "center";
+	greenAttributionContainerDiv.style.marginBottom = "2rem";
+	attributionContainerDiv.appendChild(greenAttributionContainerDiv);
+
+	// Function to update counter value and attribution text
+	const updateView = () => {
+		const greenValue = hitCounter.map?.get(greenKey);
+		const redValue = hitCounter.map?.get(redKey);
+
+		// update the counter value
+		redCounterContainerDiv.textContent = redValue.toString();
+		greenCounterContainerDiv.textContent = greenValue.toString();
+
+		const greenAttributionKey = hitCounter.map?.getAttribution(greenKey);
+		const redAttributionKey = hitCounter.map?.getAttribution(redKey);
+
+		// update the attribution text
+		const updateAttributionDisplay = (attributionKey, containerDiv) => {
+			if (attributionKey !== undefined && attributionKey.type === "op") {
```

Comment:
i'd put a comment here that a production application would want to handle detached and local attribution key types as well and render reasonable text. what you have is fine though.